### ISSUE TITLE
Meta-data editor logs out unexpectedly

### ISSUE THREAD

**Issue by matthias-ronge** [kitodo/kitodo-production#663]
During developing I recognized an annoying behaviour of the meta-data editor. The meta-data editor was already open for a while, but the last mouse clicks were just a couple of seconds ago. Suddanly, the tree-view frame turned white with red text, saying that I didn’t edit the meta-data for a long time, and that it was unlocked and now someone else has started editing it. The main frame (bottom right) was still visible. This happend on my local machine (localhost:8080). I also checked “current users” page and of course there isn’t another user.

**TL;DR The locking feature does not work as it should and needs to be fixed.**

Correct behaviour should include:
- On any interaction, the unlock countdown must newly start to count
- All frames must refer to the same session, user, and unlock counter (on the long term, get rid of frames!)
- If the meta-data becomes unlocked, but no other user starts editing them, they should immediately be locked again and the user should not be harassed
- If the meta-data cannot be locked again, the main frame should return (as when clicking “back”) and the error message should appear on the page showing then (processes, current tasks, …)
- Error message could sensibly include the display name of who has started editing the meta-data, so I can go and have a talk
- Note that `System.nanoTime()` is [not thread-safe](http://stackoverflow.com/a/9314616/1503237) and therefore cannot be used here. Stick to `System.currentTimeMillis()` which also [performs best](http://stackoverflow.com/q/19052316/1503237).

**Comment by henning-gerhardt** [comment]
Good for catching this. I see only only "problem": returning back to site before causing lose of data which are edited but not saved. But I think this is OK as without a valid (user) session saving of data is failing in most cases.

**Comment by matthias-ronge** [comment]
This is true. If the “other person” is named, it would be possible to talk and the other person logs out again, so meta-data can be locked again and save changes. Frames remain a problem but the only solution seems to me here is to get rid of the frames in whole.

**Comment by henning-gerhardt** [comment]
One other question as reproduction is time consuming: if the lock is reset inside meta-data editor then task / step is changed from "in progress" to "open" too ?

**Comment by matthias-ronge** [comment]
I had the meta-data editor open from processes list, so I cannot say. But this should be considered.

**Comment by henning-gerhardt** [comment]
I created 2 usage scenarios on behalf with a colleague and I got for me new insights:

1) I take over a step for editing metadata, opened the metadata editor and start working. My colleague searched for this process and opened the metadata editor over the process list and he starts even working without getting a message that somebody is still working on this process. **That is something which I not expected!** After closing my metadata editor I got the message that my colleague is working on this process and I should wait a few minutes. I could not open the metadata editor from "my task" list as long my colleague used the metadata editor for this process from process list.

2) We both searched for same process and start working inside metadata editor one after the next. The second user got no message that an user is working on this process.

So far as I can say that there is no check on starting metadata editor from process list if somebody already is working on this process. There must must be a check condition from "my task" list if somebody is working on this process.
